core/modules/layout/layout.admin.inc

```diff
@@ -2014,7 +2014,7 @@ function layout_block_disable_page(Layout $layout, Block $block, $renderer_name)
  *
  * @ingroup forms
  */
-function layout_condition_add_form($form, &$form_state, Layout $layout = NULL, Block $block = NULL, LayoutMenuItem $menu_item = NULL, $condition_id = NULL) {
+function layout_condition_add_form($form, &$form_state, ?Layout $layout = NULL, ?Block $block = NULL, ?LayoutMenuItem $menu_item = NULL, $condition_id = NULL) {
```

Comment:
I wonder if Layout could ever be null. Probably not a topic here in this issue, as this "null" pre-existed, but still a bit odd. The same seems true for layout_condition_edit_title().

Reply:
As with other places, we're leaving the optional and default NULL values in place where possible for now, just to get onto PHP 8.4. We could clean these up in a separate issue.

Reply:
I changed my mind here and went ahead and just made `$layout` a non-nullable required parameter, since the contents of the function may fail if `$layout` is not passed in.

Reply:
Well I tried making `$layout` required but turns out there is a valid use-case for a `NULL` layout. This same form is used to add conditions to menu items. In this situation, multiple layouts may share the same menu route, and in that situation, both `$layout` and `$block` are NULL. Only `$menu_item` has a value.

So I put this back to the way it was, with `$layout` both being optional and having a `NULL` default.